### ISSUE TITLE
OnUpdate default referential Integrity action while using Prisma to handle referential Integrity

### ISSUE THREAD

**Issue by OmG3r** [prisma/prisma#10581]
### Intro
it is mentioned in the docs that the default referential integrity actions are these here
![image](https://user-images.githubusercontent.com/10630584/144744570-075421ae-ea85-4b89-a7d0-9fcf61c1c79b.png)

but it is also stated in the docs that while using Prisma to handle referential integrity that the available actions are 
![image](https://user-images.githubusercontent.com/10630584/144744606-8e419d46-b130-44f3-9174-64cc73e134bc.png)

It is not clearly mentioned in the docs that you should manually set  the `onUpdate` to `noAction`
___________________________________________________________

### Problem
enabling the preview feature `referentialIntegrity` and attempting to update a record that has relation lead me to this error (creating and deleting records works fine)
```
Message: 
Invalid `prisma.institution.update()` invocation:


  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Server(ServerError { code: 1105, message: "symbol db.Institution.id not found", state: "HY000" })) })

Query: 
prisma.institution.update(
{
  where: {
    id: 44,
  },
  data: {
    firstName: "Random",
  },
  select: {
    id: true,
    firstName: true,
  },
}
)
```
___________________________________________________________
### Solution
setting `onUpdate` to `noAction` resolved this error. I would like  to propose to either set the default action while using `referentialIntegrity` for update operation to `noAction` (since this is the only available option at this time) or at least mention this explicitly in the documentation
___________________________________________________________
### Notes
I should also note I was using Prisma combined with Planetscale, i don't know if this is relevant but i'll just leave it here

**Comment by NBruhno** [comment]
Updating the schema with `onUpdate: NoAction` did not fix the issue initially because Prisma did not detect any schema changes when added, so it had to be resat completely. That fixed the above mentioned issue and applied the correct schema changes. I also don't remember having this issue before, but maybe I just didn't get to it until now. I'm running v3.6.0.

**Comment by dpetrick** [comment]
The docs will be clarified with the coming release. Please check back with 3.7.0 if this is clearer for you.

**Comment by janpio** [comment]
3.7.0 is out, can you take another look @NBruhno and @OmG3r please? Thanks.

**Comment by janpio** [comment]
Ok, reread the issue after spending a bit of time on the topic:

You are completely right @OmG3r, we should have disallowed anything but `NoAction` (which then of course also needs to be the default) when `referentialIntegrity=prisma` or just made the other things no ops until we implement them fully. The current situation is confusing. Because of that we have a whole other issue where multiple people reported basically the same problem: https://github.com/prisma/prisma/issues/10758 We will try to get that fixed soon.

> Updating the schema with onUpdate: NoAction did not fix the issue initially because Prisma did not detect any schema changes when added, so it had to be resat completely.

Is it possible that you did not run `npx prisma generate` to regenerate your Prisma Client after making the change? That is necessary for the Client to learn about the change in expected referential action and thus run the correct queries (or no queries at all in this case).

**Comment by NBruhno** [comment]
> Is it possible that you did not run `npx prisma generate` to regenerate your Prisma Client after making the change? That is necessary for the Client to learn about the change in expected referential action and thus run the correct queries (or no queries at all in this case).

Apologies, but I only ran `prisma migrate dev --name <name>`, (which I believe runs `generate` regardless of pending migrations or not). I think it might have mixed with the issue that I had previously with running prisma while using `pnpm` in my repo, which didn't update the client properly unless I did a reinstall of the client when doing either `migrate` or `generate`. But it looks like that is no longer an issue either (don't know which version fixed that for me, or if it was a `pnpm` issue).

I cannot reproduce my generate issue anymore. It's just the `onDelete` and `onUpdate` that does not seem to have sane defaults for `referentialIntegrity`, as you have already mentioned.
